### ISSUE TITLE
Shouldn't a Symbol with `zero=True` be constant?

### ISSUE THREAD

**Issue by JSS95** [sympy/sympy#18009]
`Symbol('x', zero=True).is_constant()` returns `False`.
But since it is nothing else but zero, isn't it supposed to return `True`?

**Comment by asmeurer** [comment]
is_constant means there are no free symbols (see its documentation), but a symbol with assumptions is still considered a free symbol, even if the assumptions only allow one actual value. Also consider

```
>>> Symbol('x', zero=True).evalf()
x
```

There are some discussions in some older issues as to why zero=True should not automatically reduce to 0 which are relevant here.

**Comment by JSS95** [comment]
> is_constant means there are no free symbols (see its documentation), but a symbol with assumptions is still considered a free symbol, even if the assumptions only allow one actual value. Also consider
> >>> Symbol('x', zero=True).evalf()
> x
> 
> There are some discussions in some older issues as to why zero=True should not automatically reduce to 0 which are relevant here.

Ah, so `is_zero` is purely assumptional property, which is equal to `nonnegative & nonpositive % real`? I can see the difference. Thanks!

Then, is `.subs()` method the only way to evaluate `Symbol('x', zero=True)` to 0?

**Comment by oscarbenjamin** [comment]
> Then, is `.subs()` method the only way to evaluate `Symbol('x', zero=True)` to 0? I expected `refine` to do it, but it didn't.

I don't understand what you want to do. `Symbol('x', zero=True)` isn't really a useful object. It can be created just because there is no reason to prevent that but why would you create that if you know that you can just use zero in the first place?

**Comment by JSS95** [comment]
> Then, is .subs() method the only way to evaluate Symbol('x', zero=True) to 0? I expected refine to do it, but it didn't.
> 
> I don't understand what you want to do. Symbol('x', zero=True) isn't really a useful object. It can be created just because there is no reason to prevent that but why would you create that if you know that you can just use zero in the first place?

You're right. I just wanted to know if we can deal with such object in assumpton-level.

Still, I think `refine` should be able to turn `Symbol('x', zero=True)` into `S.Zero`. This behavior will make such asking possible:

```
>>> ask(Q.zero(sin(x)), Q.zero(x))
# Expected True. But in current version(1.4), return None.
```

**Comment by asmeurer** [comment]
`refine` should be able to do it. `refine` currently has relatively little implemented, so that's why it doesn't do it yet.
